pkg/client/client.go

```diff
@@ -222,30 +290,71 @@ func (c *Client) Read(ctx context.Context, query string) (string, error) {
 	return string(b), nil
 }
 
-// Write initiates a write query, returning the txn hash.
-func (c *Client) Write(ctx context.Context, query string) (string, error) {
-	req := &tableland.RelayWriteQueryRequest{Statement: query}
-	var res tableland.RelayWriteQueryResponse
+type writeConfig struct {
+	relay bool
+}
+
+// WriteOption controls the behavior of Write.
+type WriteOption func(*writeConfig)
 
-	if err := c.tblRPC.CallContext(ctx, &res, "tableland_relayWriteQuery", req); err != nil {
-		return "", fmt.Errorf("calling rpc relayWriteQuery: %v", err)
+// WriteRelay specifies whether or not to relay write queries through the Tableland validator.
+// Default behavior is false for main net EVM chains, true for all others.
+func WriteRelay(relay bool) WriteOption {
+	return func(wc *writeConfig) {
+		wc.relay = relay
 	}
+}
 
-	return res.Transaction.Hash, nil
+// Write initiates a write query, returning the txn hash.
+func (c *Client) Write(ctx context.Context, query string, opts ...WriteOption) (string, error) {
+	conf := writeConfig{relay: c.relayWrites}
+	for _, opt := range opts {
+		opt(&conf)
+	}
+	if conf.relay {
+		req := &tableland.RelayWriteQueryRequest{Statement: query}
+		var res tableland.RelayWriteQueryResponse
+		if err := c.tblRPC.CallContext(ctx, &res, "tableland_relayWriteQuery", req); err != nil {
+			return "", fmt.Errorf("calling rpc relayWriteQuery: %v", err)
+		}
+		return res.Transaction.Hash, nil
+	}
+	tableID, err := c.Validate(ctx, query)
+	if err != nil {
+		return "", fmt.Errorf("calling Validate: %v", err)
+	}
+	res, err := c.tblContract.RunSQL(ctx, c.wallet.Address(), tableland.TableID(tableID), query)
+	if err != nil {
+		return "", fmt.Errorf("calling RunSQL: %v", err)
+	}
+	return res.Hash().Hex(), nil
 }
 
 // Hash validates the provided create table statement and returns its hash.
 func (c *Client) Hash(ctx context.Context, statement string) (string, error) {
 	req := &tableland.ValidateCreateTableRequest{CreateStatement: statement}
 	var res tableland.ValidateCreateTableResponse
-
 	if err := c.tblRPC.CallContext(ctx, &res, "tableland_validateCreateTable", req); err != nil {
 		return "", fmt.Errorf("calling rpc validateCreateTable: %v", err)
 	}
-
 	return res.StructureHash, nil
 }
 
+// Validate validates a write query, returning the table id.
+func (c *Client) Validate(ctx context.Context, statement string) (TableID, error) {
```

Comment:
Added support for this RPC because it is needed for making direct writes to validate the request and extract the table id from the statement.